src/Google/Task/Composer.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+use Composer\Script\Event;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class Google_Task_Composer
+{
+  public static function cleanup(Event $event)
+  {
+    $extra = $event->getComposer()->getPackage()->getExtra();
+    $servicesToKeep = $extra['google/apiclient-services'] ?? [];
```

Comment:
Defaulting this seems a bit unfriendly. If a user fat-fingered something, they'd delete all the services silently.

Additionally, I doubt this will become a problem, but we may as well sub-array the list in case we come up with another reason to use `extra`.

```json
{
	"google/apiclient-services": {
		"keep": [
			"YouTube"
		]
	}
}
```

We could use an empty array as a signal to delete all. If `extra."google/apiclient-services".keep` doesn't exist, raise a warning or error.

Reply:
also `??` doesn't work for php 5.

Reply:
I've added service validation so it *shouldnt* be possible to remove all services

Reply:
I've added tests to confirm this

Reply:
TBH I am not a fan of the "keep" identifier. I think we can exclude it for now and we could always whitelist keys like that later.